gkr_iop/src/gkr/mock.rs

```diff
@@ -147,13 +147,14 @@ impl<E: ExtensionField> MockProver<E> {
 impl<E: ExtensionField> EvalExpression<E> {
     pub fn mock_evaluate<'a>(
         &self,
-        evals: &[FieldType<'a, E>],
+        evals: &[&FieldType<'a, E>],
         challenges: &[E],
+        num_vars: usize,
     ) -> FieldType<'a, E> {
         match self {
-            EvalExpression::Zero => FieldType::default(),
+            EvalExpression::Zero => FieldType::zero(num_vars),
             EvalExpression::Single(i) => evals[*i].clone(),
-            EvalExpression::Linear(i, c0, c1) => Arc::into_inner(wit_infer_by_expr(
+            EvalExpression::Linear(i, c0, c1) => wit_infer_by_expr(
```

Comment:
```rust
match Arc::try_unwrap(wit_infer_by_expr(...)) {
        Ok(inner) => inner,
        Err(shared) => (*shared).clone(),
}
```
with that, we can do optional clone and better than current version

Reply:
👍👍👍 Fixed in https://github.com/scroll-tech/ceno/pull/970/commits/5682bd995729ad270e9a4995fbf3f6880ad9ec0b.